src/node_jsvmapi.cc

```diff
@@ -316,17 +522,47 @@ napi_value napi_create_constructor_for_wrap_with_methods(
           v8::NewStringType::kInternalized).ToLocalChecked();
   tpl->SetClassName(namestring);
 
-  for (int i = 0; i < methodcount; i++) {
-    v8::Local<v8::FunctionTemplate> t =
-        v8::FunctionTemplate::New(isolate, v8impl::FunctionCallbackWrapper,
-                                  v8::External::New(isolate,
-                                  reinterpret_cast<void*>(methods[i].callback)),
-                                  v8::Signature::New(isolate, tpl));
-    v8::Local<v8::String> fn_name =
-        v8::String::NewFromUtf8(isolate, methods[i].utf8name,
-            v8::NewStringType::kInternalized).ToLocalChecked();
-    tpl->PrototypeTemplate()->Set(fn_name, t);
-    t->SetClassName(fn_name);
+  for (int i = 0; i < property_count; i++) {
+    napi_property_descriptor* p = properties + i;
+    v8::Local<v8::String> propertyname =
+      v8::String::NewFromUtf8(isolate, p->utf8name,
+        v8::NewStringType::kInternalized).ToLocalChecked();
+
+    v8::PropertyAttribute attributes =
+      static_cast<v8::PropertyAttribute>(p->attributes);
+
+    // This code is similar to that in napi_define_property(); the
+    // difference is it applies to a template instead of an object.
+    if (p->method) {
+      v8::Local<v8::Object> cbdata = v8impl::CreateFunctionCallbackData(
+        e, p->method, p->data);
+
+      v8::Local<v8::FunctionTemplate> t = v8::FunctionTemplate::New(
+        isolate, v8impl::FunctionCallbackWrapper::Invoke, cbdata,
+        v8::Signature::New(isolate, tpl));
+      t->SetClassName(propertyname);
+
+      tpl->PrototypeTemplate()->Set(propertyname, t, attributes);
+    }
+    else if (p->getter || p->setter) {
+      v8::Local<v8::Object> cbdata = v8impl::CreateAccessorCallbackData(
+        e, p->getter, p->setter, p->data);
+
+      tpl->PrototypeTemplate()->SetAccessor(
+        propertyname,
+        v8impl::GetterCallbackWrapper::Invoke,
+        p->setter ? v8impl::SetterCallbackWrapper::Invoke : nullptr,
+        cbdata,
+        v8::AccessControl::DEFAULT,
+        attributes);
+    }
+    else {
```

Comment:
I was assuming we should preserve whatever is the existing behavior of PrototypeTemplate()->Set() when it is given a null value.